File: examples/rails/050_s3/app/controllers/s3_file_controller.rb
class S3FileController < ActionController::API
  def download
    begin
      file = S3File.find(0)
    rescue ActiveRecord::RecordNotFound => e
      render plain: "You haven't uploaded anything yet.\n" and return
    end

    send_data file.file.download, filename: file.file.blob.filename.to_s, type: file.file.blob.content_type, disposition: "inline"
  end

  def upload
    unless params.dig(:file).present?
      render plain: "You didn't pass the file to upload :(\n" and return
    end

    begin
      file = S3File.find(0)
    rescue ActiveRecord::RecordNotFound => e
      file = S3File.new
    end
    file.update(id: 0)
    file.file.attach(params[:file])
    file.save()

    render plain: "File uploaded.\n"
  end
end

File: examples/laravel/050_s3/app/Http/Controllers/S3FileController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\Storage;

class S3FileController extends Controller
{
    private $fileName = "filename";

    public function download()
    {
        if (Storage::disk('s3')->missing($this->fileName)) {
            return "You haven't uploaded anything yet.";

        }

        return Storage::disk('s3')->download($this->fileName);
    }

    public function upload(Request $request): string
    {
        if (!$request->hasFile('file')) {
            return "You didn't pass the file to upload :(";
        }

        Storage::disk('s3')->putFileAs(
            '', $request->file('file'), $this->fileName
        );

        return "File uploaded.";
    }
}
